### DISCUSSION TITLE
Tasmota power plug disconnects from wifi

### DISCUSSION THREAD

**Discussion by JIVeer** [arendst/Tasmota#18930] [Q&A]
Hi,
I am looking for some way how to investigate disconnecting device from wifi.

Device - NOUS A1T power plug (EU)
Tasmota version - 12.5.0 

Device connect to wifi fine, I can control device but after some time the device disconnects from wifi and is not available. Is not accessible by http and also MQTT connection is disconnected. I cannot find the reason why. After some time it randomly reconnects or when I reboot the router.

Original version  was 12.1.0? I made OTA already but it did not fix the problem.

Device is very close to router, I have another different tasmota plug (Shelly plug S ) with version 10.0.0 which works flawlessly.  Wifi channel is 8.

Thanks for any hints.

**Comment by hw-schrauber** [comment]
What do you want? When the device isn't available on the Wi-Fi, then there is no way to do anything on the device from outside the device.
Maybe scripting is a way to do something when the device lost the connection to the Wi-Fi. But for using scripts, you have to compile your own FW. with scripts activated.
When scripts are activated, you can control Wi-Fi up to the point to restart the device.

**Comment by JIVeer** [comment]
I want the device to be connected to the wifi always. As it disconnects randomly I need to investigate why and looking for some ideas what to check.

**Reply by hw-schrauber** [reply]
Please think about it again. If your Wi-Fi transmitter disconnects because there is a thermal problem, debugging is only possible through TASMOTA at the device. But then the output can only be done via the UART- port. You can't get any data out via WLAN.
To automate something, e.g. output of a debug, you need scripting.
Because you don't know what exactly is happening, you don't know what exactly is excluding you from the device.

**Reply by JIVeer** [reply]
I can see the console when device reconnects to wifi, it says connecting to wifi and then there is message about connection timeout. But not sure if router or device is the problem.
What script would help in this case? What should it log?

**Reply by barbudor** [reply]
Just to be sure to understand : are you seeing that on a serial console ?
Because if you were seing that on the web console, that would mean that your device is not disconnected from Wifi.
Some capture of logs may be usefull to full understand what you describe, preferable with a log level of 3

An example rule that I'm using on site where Wifi and Internet quality is very poor (the MQTT broker is cloud based):
```
Rule1 
  on system#init do RuleTimer8 %mem8% endon 
  on mqtt#disconnected do RuleTimer8 %mem8% endon 
  on mqtt#connected do RuleTimer8 0 endon 
  on Rules#Timer=8 do restart 1 endon
```
with `mem8` being the timeout, typically 5 minutes (`mem8 300`)

If you are not using MQTT you can use `Wifi#connected` and `wifi#disconncted`

**Reply by JIVeer** [reply]
I saw the message in web console after it connected to wifi again. I dont have serial attached as I did not take device apart.

**Comment by barbudor** [comment]
Do you have other Tasmota devices that do not show the same behavior?

What is your router brand/model?
ESP used to be some problem with Asus router's but AFAIK this may have been solved with recent versions so be sure to be on 12.5.*

Beside I have quite a few Gosund SP111 which are the same as the A1T. Some are quite OK, some are agging with difficulties... More disconnections, lower wifi level despite being at the same location as another that is fine...

**Reply by JIVeer** [reply]
I have also Shelly Plug S which is on older tasmota version and works fine.
Router is TP-Link AX20.

**Reply by Jason2866** [reply]
"It's dead Jim" Cheap components are dying fast.
